Added a second commit for `useClangAssembler` and `useLLVMArchiver`.

The reason is that I'm having the following error when building `nixStatic`, which seems to disappear when setting `useLLVMArchiver = !stdenv.targetPlatform.isStatic`. 😕

```
-- Check for working C compiler: /nix/store/rmw9yzrp2gvr44q76qjbx7czb1swa4z6-aarch64-apple-darwin-clang-wrapper-11.1.0/bin/aarch64-apple-darwin-clang
-- Check for working C compiler: /nix/store/rmw9yzrp2gvr44q76qjbx7czb1swa4z6-aarch64-apple-darwin-clang-wrapper-11.1.0/bin/aarch64-apple-darwin-clang - broken
CMake Error at /nix/store/fgqy9w99yf2r4fr5i6amlyl665rci4vw-cmake-3.26.4/share/cmake-3.26/Modules/CMakeTestCCompiler.cmake:67 (message):
  The C compiler

    "/nix/store/rmw9yzrp2gvr44q76qjbx7czb1swa4z6-aarch64-apple-darwin-clang-wrapper-11.1.0/bin/aarch64-apple-darwin-clang"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /tmp/nix-build-compiler-rt-libc-static-aarch64-apple-darwin-11.1.0.drv-2/compiler-rt-11.1.0.src/build/CMakeFiles/CMakeScratch/TryCompile-abtxjZ

    Run Build Command(s):/nix/store/fgqy9w99yf2r4fr5i6amlyl665rci4vw-cmake-3.26.4/bin/cmake -E env VERBOSE=1 /nix/store/g92m18s55sagq61zc2a9jkdr7s6fifkr-gnumake-4.4.1/bin/make -f Makefile cmTC_36272/fast && /nix/store/g92m18s55sagq61zc2a9jkdr7s6fifkr-gnumake-4.4.1/bin/make  -f CMakeFiles/cmTC_36272.dir/build.make CMakeFiles/cmTC_36272.dir/build
    make[1]: Entering directory '/private/tmp/nix-build-compiler-rt-libc-static-aarch64-apple-darwin-11.1.0.drv-2/compiler-rt-11.1.0.src/build/CMakeFiles/CMakeScratch/TryCompile-abtxjZ'
    Building C object CMakeFiles/cmTC_36272.dir/testCCompiler.c.o
    /nix/store/rmw9yzrp2gvr44q76qjbx7czb1swa4z6-aarch64-apple-darwin-clang-wrapper-11.1.0/bin/aarch64-apple-darwin-clang --target=aarch64-apple-darwin   -arch arm64 -MD -MT CMakeFiles/cmTC_36272.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_36272.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_36272.dir/testCCompiler.c.o -c /tmp/nix-build-compiler-rt-libc-static-aarch64-apple-darwin-11.1.0.drv-2/compiler-rt-11.1.0.src/build/CMakeFiles/CMakeScratch/TryCompile-abtxjZ/testCCompiler.c
    Linking C static library libcmTC_36272.a
    /nix/store/fgqy9w99yf2r4fr5i6amlyl665rci4vw-cmake-3.26.4/bin/cmake -P CMakeFiles/cmTC_36272.dir/cmake_clean_target.cmake
    /nix/store/fgqy9w99yf2r4fr5i6amlyl665rci4vw-cmake-3.26.4/bin/cmake -E cmake_link_script CMakeFiles/cmTC_36272.dir/link.txt --verbose=1
    "" qc libcmTC_36272.a CMakeFiles/cmTC_36272.dir/testCCompiler.c.o
    Error running link command: No such file or directory
    make[1]: *** [CMakeFiles/cmTC_36272.dir/build.make:101: libcmTC_36272.a] Error 2
    make[1]: Leaving directory '/private/tmp/nix-build-compiler-rt-libc-static-aarch64-apple-darwin-11.1.0.drv-2/compiler-rt-11.1.0.src/build/CMakeFiles/CMakeScratch/TryCompile-abtxjZ'
    make: *** [Makefile:127: cmTC_36272/fast] Error 2





  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:14 (project)


-- Configuring incomplete, errors occurred!
error: builder for '/nix/store/2lcagrvz7xy45w9mbbgh8jar70jj0arf-compiler-rt-libc-static-aarch64-apple-darwin-11.1.0.drv' failed with exit code 1;
```

--- a/pkgs/os-specific/darwin/cctools/llvm.nix
+++ b/pkgs/os-specific/darwin/cctools/llvm.nix
@@ -1,9 +1,11 @@
 # Create a cctools-compatible bintools that uses equivalent tools from LLVM in place of the ones
 # from cctools when possible.
 
-{ lib, stdenv, makeWrapper, cctools-port, llvmPackages, enableManpages ? true }:
+{ lib, stdenv, makeWrapper, cctools-port, llvmPackages, enableManpages ? stdenv.targetPlatform == stdenv.hostPlatform }:
 
 let
+  inherit (stdenv) targetPlatform hostPlatform;
+
   cctoolsVersion = lib.getVersion cctools-port;
   llvmVersion = llvmPackages.release_version;
 
@@ -52,7 +54,7 @@ let
   ++ lib.optional (!useLLVMOtool) "otool"
   ++ lib.optional (!useLLVMStrip) "strip";
 
-  inherit (stdenv.cc) targetPrefix;
+  targetPrefix = lib.optionalString (targetPlatform != hostPlatform) "${targetPlatform.config}-";
 
   linkManPages = pkg: source: target: lib.optionalString enableManpages ''
     sourcePath=${pkg}/share/man/man1/${source}.1.gz
@@ -81,7 +83,7 @@ stdenv.mkDerivation {
     makeWrapper "${lib.getBin llvmPackages.clang-unwrapped}/bin/clang" "$out/bin/${targetPrefix}as" \
       --add-flags "-x assembler -integrated-as -c"
 
-    ln -s "${lib.getBin llvmPackages.bintools-unwrapped}/bin/llvm-ar" "$out/bin/${targetPrefix}ar"
+    ln -s "${lib.getBin llvmPackages.bintools-unwrapped}/bin/${targetPrefix}llvm-ar" "$out/bin/${targetPrefix}ar"
     ${linkManPages llvmPackages.llvm-manpages "llvm-ar" "ar"}
 
     for tool in ${toString llvm_bins}; do

